test/server/lib/validation/form_validators/ManageApplications.test.js

```diff
@@ -0,0 +1,149 @@
+
+  it('should validate the addApplication sucessfully and return true', () => {
+    const payload = {
+      applicationName: 'hi',
+      terms: 'https://eg.com',
+      privacyPolicy: 'https://eg.com',
+      environment: 'test'
+    };
+    const stubValidName = sinon.stub(Validator.prototype, 'isValidAppName').returns(true);
+    const stubValidDesc = sinon.stub(Validator.prototype, 'isValidDescription').returns(true);
+    const stubValidUrl = sinon.stub(Validator.prototype, 'isValidUrl').returns(true);
+    const stubKeys = sinon.stub(ManageApplication.prototype, '_formatIncomingPayload').returns(routeData.addApplication);
+    expect(ManageApplication.prototype.addApplication(payload)).to.eventually.equal(true);
+    expect(stubValidName).to.have.been.calledOnce;
+    expect(stubValidDesc).to.have.been.calledOnce;
+    expect(stubKeys).to.have.been.calledOnce;
+    expect(stubValidUrl).to.have.been.calledTwice;
+    expect(stubLogger).to.have.been.calledOnce;
+  });
+  it('should validate the addApplication and return invalid environment validation error', () => {
+    const payload = {
+      applicationName: 'name',
+      terms: '',
+      privacyPolicy: '',
+      environment: ''
+    };
+    const validationException = exceptions.validationException;
+    const stubValidName = sinon.stub(Validator.prototype, 'isValidAppName').returns(true);
+    const stubValidDesc = sinon.stub(Validator.prototype, 'isValidDescription').returns(true);
+    const stubKeys = sinon.stub(ManageApplication.prototype, '_formatIncomingPayload').returns(routeData.addApplication);
+    expect(ManageApplication.prototype.addApplication(payload)).to.be.rejectedWith(validationException);
+    expect(stubValidName).to.have.been.calledOnce;
+    expect(stubValidDesc).to.have.been.calledOnce;
+    expect(stubKeys).to.have.been.calledOnce;
+    expect(stubLogger).to.have.been.calledTwice;
+  });
+
+  it('should validate the addApplication and return blank name validation error, ', () => {
+    const payload = {
+      applicationName: '',
+      terms: '',
+      privacyPolicy: '',
+      environment: 'test'
+    };
+    const validationException = exceptions.validationException;
+    const stubValidName = sinon.stub(Validator.prototype, 'isValidAppName').returns(false);
+    const stubValidDesc = sinon.stub(Validator.prototype, 'isValidDescription').returns(true);
+    const stubKeys = sinon.stub(ManageApplication.prototype, '_formatIncomingPayload').returns(routeData.addApplication);
+    expect(ManageApplication.prototype.addApplication(payload)).to.be.rejectedWith(validationException);
+    expect(stubValidName).to.have.been.calledOnce;
+    expect(stubValidDesc).to.have.been.calledOnce;
+    expect(stubKeys).to.have.been.calledOnce;
+    expect(stubLogger).to.have.been.calledTwice;
+  });
+  it('should validate the addApplication and return invalid name validation error, ', () => {
+    const payload = {
+      applicationName: ')))))(((((******INVALID_NAME',
```

Comment:
Why are we setting actual values if we just stub the method to fail?

Reply:
I did that to make it more readable, it might just be me that finds it that way